### ISSUE TITLE
Events not sent to Segment servers: The operation was canceled.

### ISSUE THREAD

**Issue by mabead** [segmentio/Analytics.NET#179]
Following the fix provided in https://github.com/segmentio/Analytics.NET/issues/167, we have much fewer lost messages. But once in a while (once every few days), we get errors like this:

```
Failed to send analytic event track for user b5032b9d-51be-420d-877b-32f829d0be6e and message ab25555b-d5e7-437e-b5d4-f4522bb09858 to segment.io server. Details: The operation was canceled.
System.OperationCanceledException: The operation was canceled.
   at System.Threading.CancellationToken.ThrowOperationCanceledException()
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.ThrowException(SocketError error, CancellationToken cancellationToken)
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.GetResult(Int16 token)
   at System.Net.FixedSizeReader.ReadPacketAsync(Stream transport, AsyncProtocolRequest request)
   at System.Net.Security.SslStream.EndProcessAuthentication(IAsyncResult result)
   at System.Threading.Tasks.TaskFactory`1.FromAsyncCoreLogic(IAsyncResult iar, Func`2 endFunction, Action`1 endAction, Task`1 promise, Boolean requiresSynchronization)
--- End of stack trace from previous location where exception was thrown ---
   at System.Net.Http.ConnectHelper.EstablishSslConnectionAsyncCore(Stream stream, SslClientAuthenticationOptions sslOptions, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.ConnectAsync(HttpRequestMessage request, Boolean allowHttp2, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.CreateHttp11ConnectionAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.GetHttpConnectionAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.SendWithRetryAsync(HttpRequestMessage request, Boolean doRequestAuth, CancellationToken cancellationToken)
   at System.Net.Http.RedirectHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Net.Http.DiagnosticsHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Net.Http.HttpClient.FinishSendAsyncBuffered(Task`1 sendTask, HttpRequestMessage request, CancellationTokenSource cts, Boolean disposeCts)
   at Segment.Request.BlockingRequestHandler.MakeRequest(Batch batch)
```

This message is logged by our delegate that is hooked on `Segment.Client.Failed`. 

I think that you may be missing a `catch` on `OperationCanceledException` here https://github.com/segmentio/Analytics.NET/blob/63ba5d1f90bf79c498088e8c008e42c0486f4767/Analytics/Request/BlockingRequestHandler.cs#L278.

From a different perspective, maybe it would be safer to retry (instead of dropping) messages when `PostAsync` throws an unexpected exception.

**Comment by juliette-ruchel** [comment]
Hi @mabead,

We have looked into your problem, and we have added handling for the OperationCanceledException and Excpetion in general. We decided to keep it in two separated catches so when you have an exception like these is easier to understand possible problems that might be happening in case retrying sending the message fails.

**Comment by mabead** [comment]
Thanks for the fix @juliette-ruchel. @pooyaj when do you plan to publish a new version of the library?

**Comment by mabead** [comment]
The bug reported here was related to version 3.8.0 of `Analytics.NET`. I see that no new version is yet available. @juliette-ruchel @pooyaj , do you plan on releasing the fix soon?

**Comment by mabead** [comment]
Hi @juliette-ruchel and @pooyaj . I'm still waiting for that fix to be available in a nuget package. Thanks.

**Comment by pooyaj** [comment]
Hi @mabead, Looks like #181 was supposed to fix this issue, and it was included in 3.8.0. Are you still seeing the issue? If so we probably have to re-open this, and work on another fix.

**Comment by mabead** [comment]
@pooyaj my bad. It looks like you're right. With 3.8.0, it looks like the problem is fixed. Thanks for your help and sorry for the wrong information.
